props/saber_caiwyn_buttons.h

```diff
@@ -369,24 +368,22 @@ class CaiwynButtons : public PropBase {
   }
 
   void NextTrack(bool manually_selected) {
-    if (manually_selected) {
-      track_player_->Stop();
-      track_player_.Free();
-#ifdef CAIWYN_SAVE_TRACKS
-      current_preset_.track = mkstr(current_track_);
-      current_preset_.Save();
-#endif
-      if (SFX_mselect) {
-        hybrid_font.PlayPolyphonic(&SFX_mselect);
-      }
-    }
+    if (track_player_->isPlaying()) track_player_->Stop();
+    if (track_player_) track_player_.Free();
```

Comment:
why free the track player?
Aren't you just going to use it again anyways?

Reply:
Good question.  Mainly because PlayTrack() invokes GetFreeWavPlayer() but I suppose I could move that elsewhere... let me see what I can do with that.

Reply:
When a track finishes playing on its own, does the track player get freed automatically?  I've built my track player on the assumption that it does...

Reply:
The track player is a RefPtr<>, which would normally mean that the answer is no, it does not get freed automatically. However there is a bit of code in PropBase::Loop() that frees the track player once it stops playing.